plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -944,6 +946,321 @@ public void testSettingChangeDataFeedEnabledProperty()
+                " (id BIGINT, nested1 ROW(a BIGINT, b VARCHAR, c INT), nested2 ROW(d DOUBLE, e BOOLEAN, f DATE))");
+        assertUpdate("INSERT INTO " + tableName1 + " VALUES" +
+                        " (100, ROW(10, 'a', 100), ROW(10.10, true, DATE '2023-04-19'))",
+                1);
+
+        assertUpdate("CREATE TABLE " + tableName2 +
+                " (nested2 ROW(d DOUBLE, e BOOLEAN, f DATE), id BIGINT, nested1 ROW(a BIGINT, b VARCHAR, c INT))");
+        assertUpdate("INSERT INTO " + tableName2 + " VALUES" +
+                        " (ROW(10.10, true, DATE '2023-04-19'), 100, ROW(10, 'a', 100))",
+                1);
+
+        String table1DataFile = getTableFiles(tableName1).stream()
+                .filter(file -> !file.contains("/_delta_log/"))
+                .findAny()
+                .orElseThrow();
+        table1DataFile = table1DataFile.replaceFirst("s3://" + BUCKET_NAME, "");
+
+        String table2DataFile = getTableFiles(tableName2).stream()
+                .filter(file -> !file.contains("/_delta_log/"))
+                .findAny()
+                .orElseThrow();
+        table2DataFile = table2DataFile.replaceFirst("s3://" + BUCKET_NAME, "");
+
+        MinioClient client = hiveMinioDataLake.getMinioClient();
+        // Replace table1 data file with table2 data file, so that the table's schema and data's schema has different column order
+        client.copyObject(BUCKET_NAME, table2DataFile, BUCKET_NAME, table1DataFile);
+
+        assertQuery("SELECT nested2.e, nested1.a, nested2.f, nested1.b, id FROM " + tableName1, "VALUES (true, 10, DATE '2023-04-19', 'a', 100)");
+        assertQuery("SELECT nested2.e, nested1.a, nested2.f, nested1.b, id FROM " + tableName2, "VALUES (true, 10, DATE '2023-04-19', 'a', 100)");
+
+        assertUpdate("DROP TABLE " + tableName1);
+        assertUpdate("DROP TABLE " + tableName2);
+    }
+
+    @Test
+    public void testProjectionPushdownWithCdfEnabled()
+    {
+        String tableName = "test_projection_pushdown_with_cdf_enabled_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (id BIGINT, root ROW(f1 BIGINT, f2 BIGINT)) WITH (change_data_feed_enabled = true)");
+        assertUpdate("INSERT INTO " + tableName + " VALUES (1, ROW(1, 2)), (1, ROW(2, 3)), (1, ROW(3, 4))", 3);
+        assertQuery("SELECT id, root.f2 FROM " + tableName, "VALUES (1, 2), (1, 3), (1, 4)");
+
+        assertUpdate("ALTER TABLE " + tableName + " SET PROPERTIES change_data_feed_enabled = false");
+        assertUpdate("INSERT INTO " + tableName + " VALUES (1, ROW(4, 5))", 1);
+        assertQuery("SELECT id, root.f2 FROM " + tableName, "VALUES (1, 2), (1, 3), (1, 4), (1, 5)");
+
+        assertUpdate("ALTER TABLE " + tableName + " SET PROPERTIES change_data_feed_enabled = true");
+        assertUpdate("INSERT INTO " + tableName + " VALUES (1, ROW(5, 6))", 1);
+        assertQuery("SELECT id, root.f2 FROM " + tableName, "VALUES (1, 2), (1, 3), (1, 4), (1, 5), (1, 6)");
```

Comment:
Why is this test case relevant?

Reply:
I wanted to verify the case with cdf feature enable.